Parser/pegen/peg_api.c

```diff
@@ -4,25 +4,35 @@
 #include "pegen.h"
 
 mod_ty
-PyPegen_ASTFromString(const char *str, PyArena *arena)
+PyPegen_ASTFromString(const char *str, PyObject *filename_ob, PyArena *arena)
 {
-    return run_parser_from_string(str, parse_start, RAW_AST_OBJECT, arena);
+    return run_parser_from_string(str, parse_start, filename_ob, arena);
 }
 
 mod_ty
-PyPegen_ASTFromFile(const char *filename, PyArena *arena)
+PyPegen_ASTFromFile(const char *filename, PyObject *filename_ob, PyArena *arena)
 {
-    return run_parser_from_file(filename, parse_start, RAW_AST_OBJECT, arena);
+    return run_parser_from_file(filename, parse_start, filename_ob, arena);
 }
 
 PyCodeObject *
-PyPegen_CodeObjectFromString(const char *str, PyArena *arena)
+PyPegen_CodeObjectFromString(const char *str, PyObject *filename_ob, PyArena *arena)
```

Comment:
Why does this need the caller to create and free the arena? Once we're done generating the code we don't need it any more. (This is different for the previous two.)

Reply:
No need to, I just did this like that in order to be consistent with `PyAST_CompileObject` which expects the arena as an argument.